dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/storage/S3Store.java

```diff
@@ -0,0 +1,98 @@
+package com.salesforce.dockerfileimageupdate.storage;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.ListObjectsV2Result;
+import com.amazonaws.services.s3.model.S3Object;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+import com.salesforce.dockerfileimageupdate.utils.DockerfileGitHubUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+public class S3Store implements ImageTagStore {
+    private static final Logger log = LoggerFactory.getLogger(S3Store.class);
+    private static final Character S3_FILE_KEY_PATH_DELIMITER = ';';
```

Comment:
See https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html .  I don't think `;` is a safe character of object names.

Reply:
Great find! I recommend going with `.`

Reply:
@afalko @cjneasbi Using `.` will make it difficult to regenerate the correct image name when we read the content from the S3 bucket since the filename for an image `dva-registry.internal.salesforce.com/sfci/sfci-dfiu-test/dfiu-test-multiple-from` will now look like this 
`dva-registry.internal.salesforce.com.sfci.sfci-dfiu-test.dfiu-test-multiple-from`. I suggest using `!` as the delimiter since I have not seen that character being used in org, repo or image name. Thoughts?

Reply:
Good point! `!` seems good to me